homeassistant/components/mill/config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""Adds config flow for Mill integration."""
+import logging
+
+from mill import Mill
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class MillConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Mill integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors={},
+            )
+
+        username = user_input[CONF_USERNAME].replace(" ", "")
+        password = user_input[CONF_PASSWORD].replace(" ", "")
+
+        mill_data_connection = Mill(
+            username, password, websession=async_get_clientsession(self.hass),
+        )
+
+        errors = {}
+
+        if not await mill_data_connection.connect():
```

Comment:
Can this ever throw an exception?

Reply:
I catch the most common errors here:
https://github.com/Danielhiversen/pymill/blob/master/mill/__init__.py#L72
Might be others?

Reply:
If its never going to happen then probably ok as-is.

The usual pattern looks something like this.
```

            try:
                .....
            except InvalidAuth:
                errors["base"] = "invalid_auth"
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"
```

Reply:
Just noticed there is no `.disconnect()` called.  Is this going to leave behind an open connection?

Reply:
No, Instead of `connect` the name of the function should maybe have been `login`

It does a websession.post with the websession from `async_get_clientsession(hass)`